pkg/apis/core/validation/validation.go

```diff
@@ -2460,9 +2455,7 @@ func ValidatePersistentVolumeClaimUpdate(newPvc, oldPvc *core.PersistentVolumeCl
 		}
 		if opts.EnableVolumeAttributesClass {
 			if oldPvc.Spec.VolumeAttributesClassName != nil {
-				if newPvc.Spec.VolumeAttributesClassName == nil {
-					allErrs = append(allErrs, field.Forbidden(field.NewPath("spec", "volumeAttributesClassName"), "update from non-nil value to nil is forbidden"))
-				} else if len(*newPvc.Spec.VolumeAttributesClassName) == 0 {
+				if newPvc.Spec.VolumeAttributesClassName != nil && len(*newPvc.Spec.VolumeAttributesClassName) == 0 {
```

Comment:
I'm looking at the api description for vacName here: https://github.com/kubernetes/kubernetes/blob/13dcec3d1bb6e76526caa0c198226605ffe62ce9/pkg/apis/core/types.go#L565-L568

Do we actually support a default VAC? cc @sunnylovestiramisu 

Also is there a difference between nil and empty string?

Reply:
No we do not, the argument is that default VAC is without a VAC.

An empty string value means that no VolumeAttributesClass, but once it is set to a string, we do not allow to set it back to empty string.

The API description needs to be updated in https://github.com/kubernetes/kubernetes/pull/131549/files